diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt

```diff
@@ -46,32 +46,35 @@ class PackageNaming(private val configRules: List<RulesConfig>) : Rule("package-
-        domainName = configuration.also {
+        configuration.also {
             if (it.isDefault && visitorCounter.incrementAndGet() == 1) {
-                log.error("Not able to find an external configuration for domain name in the common" +
-                        " configuration (is it missing in yml config?)")
+                log.error("Not able to find configuration file")
             }
         }
-            .domainName
-
-        if (node.elementType == PACKAGE_DIRECTIVE) {
-            val filePath = node.getRootNode().getFilePath()
-            // calculating package name based on the directory where the file is placed
-            val realPackageName = calculateRealPackageName(filePath)
-
-            // if node isLeaf - this means that there is no package name declared
-            if (node.isLeaf() && !filePath.isKotlinScript()) {
-                warnAndFixMissingPackageName(node, realPackageName, filePath)
-                return
-            }
-
-            // getting all identifiers from existing package name into the list like [org, diktat, project]
-            val wordsInPackageName = node.findAllNodesWithSpecificType(IDENTIFIER)
-
-            // no need to check that packageIdentifiers is empty, because in this case parsing will fail
-            checkPackageName(wordsInPackageName, node)
-            // fix in checkFilePathMatchesWithPackageName is much more aggressive than fixes in checkPackageName, they can conflict
-            checkFilePathMatchesWithPackageName(wordsInPackageName, realPackageName, node)
+            .domainName?.let {
+                domainName = it
+                if (node.elementType == PACKAGE_DIRECTIVE) {
+                    val filePath = node.getRootNode().getFilePath()
+                    // calculating package name based on the directory where the file is placed
+                    val realPackageName = calculateRealPackageName(filePath)
+
+                    // if node isLeaf - this means that there is no package name declared
+                    if (node.isLeaf() && !filePath.isKotlinScript()) {
+                        warnAndFixMissingPackageName(node, realPackageName, filePath)
+                        return
+                    }
+
+                    // getting all identifiers from existing package name into the list like [org, diktat, project]
+                    val wordsInPackageName = node.findAllNodesWithSpecificType(IDENTIFIER)
+
+                    // no need to check that packageIdentifiers is empty, because in this case parsing will fail
+                    checkPackageName(wordsInPackageName, node)
+                    // fix in checkFilePathMatchesWithPackageName is much more aggressive than fixes in checkPackageName, they can conflict
+                    checkFilePathMatchesWithPackageName(wordsInPackageName, realPackageName, node)
+                }
+            } ?: if (visitorCounter.get() == 1) {
```

Comment:
I think it's better to use `if-else` instead of `?.let {} ?:` here, we specifically allowed in our code style using `if-else` when there are complex conditions.

Reply:
We have got `AVOID_NULL_CHECKS` rule, that will be trigger if I change it.